OpenRA.Mods.Common/Traits/BotModules/BuildingRepairBotModule.cs

```diff
@@ -0,0 +1,42 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Mods.Common.AI;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Manages AI repairing base buildings.")]
+	public class BuildingRepairBotModuleInfo : ConditionalTraitInfo
+	{
+		public override object Create(ActorInitializer init) { return new BuildingRepairBotModule(init.Self, this); }
+	}
+
+	public class BuildingRepairBotModule : ConditionalTrait<BuildingRepairBotModuleInfo>, IBotRespondToAttack
+	{
+		public BuildingRepairBotModule(Actor self, BuildingRepairBotModuleInfo info)
+			: base(info) { }
+
+		void IBotRespondToAttack.RespondToAttack(IBot bot, Actor self, AttackInfo e)
```

Comment:
Do we need to worry about repairs being automatically cancelled by something else?

Reply:
No idea.
But unless I'm missing something, this shouldn't break anything that wasn't already broken on bleed, at least.